crates/worker/src/partition/state_machine/lifecycle/notify_sleep_completion.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2023 - 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::debug_if_leader;
+use crate::partition::state_machine::invocation_status_ext::InvocationStatusExt;
+use crate::partition::state_machine::{CommandHandler, Error, StateMachineApplyContext, entries};
+use restate_storage_api::fsm_table::FsmTable;
+use restate_storage_api::invocation_status_table::{InvocationStatus, InvocationStatusTable};
+use restate_storage_api::journal_table as journal_table_v1;
+use restate_storage_api::journal_table_v2;
+use restate_storage_api::outbox_table::OutboxTable;
+use restate_storage_api::promise_table::PromiseTable;
+use restate_storage_api::state_table::StateTable;
+use restate_storage_api::timer_table::TimerTable;
+use restate_types::identifiers::InvocationId;
+use restate_types::invocation::InvocationEpoch;
+use restate_types::journal_v2::{CompletionId, SleepCompletion};
+
+pub struct OnNotifySleepCompletionCommand {
```

Comment:
Now I am even more confused wrt to `OnNotifyInvocationResponse`.

Reply:
See comment above